### ISSUE TITLE
Some test cases in IoUtilTest class fails on Macos

### ISSUE THREAD

**Issue by arman-yekkehkhani** [eclipse-jkube/jkube#3001]
### Describe the bug

When running tests in `org.eclipse.jkube.kit.common.util.IoUtilTest`, some test cases fail with a message similar to the following one:

```
java.lang.AssertionError: 
Expecting actual:
  ["../../../../../../private/var/folders/5q/fb9_qmc91h76mqsnyqxnf7x00000gn/T/junit14779572735641380464/deployment.yaml"]
to contain exactly in any order:
  ["deployment.yaml"]
elements not found:
  ["deployment.yaml"]
and elements not expected:
  ["../../../../../../private/var/folders/5q/fb9_qmc91h76mqsnyqxnf7x00000gn/T/junit14779572735641380464/deployment.yaml"]
```
After reading the implementation of method `fileTree` in class `FileAssertions`, I found out that the relative path of files in the file tree is not computed correctly on MacOS.

P.S. Fixed the issue. Shall I create a PR for it?

### Eclipse JKube version

SNAPSHOT

### Component

JKube Kit

### Apache Maven version

None

### Gradle version

None

### Steps to reproduce

1. It requires MacOS
2. Run `mvn clean test` inside `jkube-jit` module
3. Observe the error

### Expected behavior

Tests inside class `IoUtilTest` should all pass.

### Runtime

other (please specify in additional context)

### Kubernetes API Server version

1.25.3

### Environment

macOS

**Comment by rohanKanojia** [comment]
@arman-yekkehkhani : Thanks for reporting! 

> P.S. Fixed the issue. Shall I create a PR for it?

If it's a small change, go ahead and go it. If it's slightly bigger change maybe you want to discuss on this issue first.

**Comment by rohanKanojia** [comment]
@arman-yekkehkhani : Maybe we can also add a GitHub Action workflow for MacOS build. But maybe we can create separate task for this.

**Comment by arman-yekkehkhani** [comment]
@rohanKanojia
Thanks! Added the corresponding PR.
Yeah, adding a workflow for MacOS would be nice. But I am not sure how. Do I have access to that?

**Comment by rohanKanojia** [comment]
@arman-yekkehkhani : Adding a macOS workflow just means adding a GitHub Action YAML file similar to [existing workflow](https://github.com/eclipse-jkube/jkube/blob/master/.github/workflows/license.yml)s in `.github/workflows` directory. For MacOS you just need to change the runner to `runs-on: macos-latest`

**Comment by manusa** [comment]
However, this is mostly internal work. We'll take care of this ourselves, thanks!

**Comment by arman-yekkehkhani** [comment]
@rohanKanojia I have attached a PR. Could you please take a look at it?

**Comment by rohanKanojia** [comment]
@arman-yekkehkhani : Unfortunately, I don't have access to MacOS environment where I can test your fix. Could you please attach some screenshot/ GitHub Action log to verify that it's working on MacOS?

**Comment by arman-yekkehkhani** [comment]
@rohanKanojia Here are the screenshot and logs when I run tests locally for module `JKube Kit :: Common` which contains the test class.
<img width="1508" alt="Screenshot 2024-05-03 at 11 13 33" src="https://github.com/eclipse-jkube/jkube/assets/72594459/1526b867-af4e-41c5-8105-303050d10c85">

logs:
```
[INFO] Scanning for projects...
[INFO] 
[INFO] -----------------< org.eclipse.jkube:jkube-kit-common >-----------------
[INFO] Building JKube Kit :: Common 1.17-SNAPSHOT
[INFO] --------------------------------[ jar ]---------------------------------
Downloading from maven-central: https://repo1.maven.org/maven2/org/eclipse/jkube/jkube-kit-api/1.17-SNAPSHOT/maven-metadata.xml
Downloading from maven-central: https://repo1.maven.org/maven2/org/eclipse/jkube/jkube-kit-parent/1.17-SNAPSHOT/maven-metadata.xml
Downloading from maven-central: https://repo1.maven.org/maven2/org/eclipse/jkube/jkube/1.17-SNAPSHOT/maven-metadata.xml
[INFO] 
[INFO] --- maven-enforcer-plugin:3.4.1:enforce (dependency-convergence) @ jkube-kit-common ---
[INFO] Rule 0: org.apache.maven.enforcer.rules.dependency.DependencyConvergence passed
[INFO] 
[INFO] --- maven-resources-plugin:3.3.1:resources (default-resources) @ jkube-kit-common ---
[INFO] Copying 1 resource from src/main/resources to target/classes
[INFO] 
[INFO] --- maven-compiler-plugin:3.13.0:compile (default-compile) @ jkube-kit-common ---
[INFO] Nothing to compile - all classes are up to date.
[INFO] 
[INFO] --- maven-resources-plugin:3.3.1:testResources (default-testResources) @ jkube-kit-common ---
[INFO] Copying 57 resources from src/test/resources to target/test-classes
[INFO] 
[INFO] --- maven-compiler-plugin:3.13.0:testCompile (default-testCompile) @ jkube-kit-common ---
[INFO] Nothing to compile - all classes are up to date.
[INFO] 
[INFO] --- maven-surefire-plugin:3.2.5:test (default-test) @ jkube-kit-common ---
[INFO] Using auto detected provider org.apache.maven.surefire.junitplatform.JUnitPlatformProvider
[INFO] 
[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] Running org.eclipse.jkube.kit.common.util.IoUtilTest
SLF4J: Failed to load class "org.slf4j.impl.StaticLoggerBinder".
SLF4J: Defaulting to no-operation (NOP) logger implementation
SLF4J: See http://www.slf4j.org/codes.html#StaticLoggerBinder for further details.
[INFO] Tests run: 509, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 2.099 s -- in org.eclipse.jkube.kit.common.util.IoUtilTest
[INFO] 
[INFO] Results:
[INFO] 
[INFO] Tests run: 509, Failures: 0, Errors: 0, Skipped: 0
[INFO] 
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  7.220 s
[INFO] Finished at: 2024-05-03T11:18:26+03:30
[INFO] ------------------------------------------------------------------------
```

**Comment by rohanKanojia** [comment]
@arman-yekkehkhani : Hello, is this issue fixed by #3010 ? Shall we close it?

**Comment by arman-yekkehkhani** [comment]
@rohanKanojia Yeah 👍
